### PR TITLE
Implement new kv and kv rw interface

### PR DESCRIPTION
**Pull request by linse**
Depends on https://github.com/mirage/mirage-kv/pull/14 /cc @samoht 
Tests are also updated and work. We still have assert false for `digest` and `batch`.

### CODE REVIEW

File: src/mirage_fs_unix.ml [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+      | None -> Error (`Storage_error (key, "mtime parsing failed"))
+      | Some ts -> Ok (Ptime.Span.to_d_ps (Ptime.to_span ts))
+    )
+    (function
+      | Unix.Unix_error (e, _, _) -> Lwt.return (Error (`Storage_error (key, Unix.error_message e)))
+      | e -> Lwt.fail e)
+
+let batch _ ?retries:_ret _ = assert false
+
+let digest _ _ = assert false
```
Comment (samoht) [inline src/mirage_fs_unix.ml]: Maybe use `Digest.file` ?

File: src/mirage_fs_unix.ml [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+      let mtime = stat.Lwt_unix.LargeFile.st_mtime in
+      match Ptime.of_float_s mtime with
+      | None -> Error (`Storage_error (key, "mtime parsing failed"))
+      | Some ts -> Ok (Ptime.Span.to_d_ps (Ptime.to_span ts))
+    )
+    (function
+      | Unix.Unix_error (e, _, _) -> Lwt.return (Error (`Storage_error (key, Unix.error_message e)))
+      | e -> Lwt.fail e)
+
+let batch _ ?retries:_ret _ = assert false
```
Comment (samoht) [inline src/mirage_fs_unix.ml]: You can probably assume that `type batch = t` here and use https://github.com/mirage/irmin/blob/master/src/irmin-mem/irmin_mem.ml#L35-L36

File: src/mirage_fs_unix.ml (hunk 1 above)
Comment (linse) [reply src/mirage_fs_unix.ml]: We moved to https://github.com/mirage/mirage-fs-unix and will address your suggestions there.

File: src/mirage_fs_unix.ml (hunk 2 above)
Comment (linse) [reply src/mirage_fs_unix.ml]: From the code, we don't fully understand how the `batch` type/operation works in `irmin` yet. We'll use `set` (single write operations) for now, but would love to have a meeting to sketch out the `batch` approach with you in `irmin` and `mirage-kv-mem`.

**Comment by linse** [comment]
Discussion can continue on https://github.com/mirage/mirage-fs-unix

**Pull request closed without merging**